server/src/test/java/org/elasticsearch/common/xcontent/support/XContentMapValuesTests.java

```diff
@@ -479,6 +482,79 @@ public void testSharedPrefixes() {
         assertEquals(Collections.singletonMap("foobaz", 3), XContentMapValues.filter(map, new String[0], new String[] {"foobar"}));
     }
 
+    @Override
+    public void testSimpleArrayOfObjectsExclusive() throws Exception {
```

Comment:
I guess our map-based filtering will now behave differently than direct xContent filtering. This seems okay to me, since I think it already diverged -- map-based filtering preserves empty objects, whereas xContent filtering doesn't?

Reply:
yes, preserving empty objects was until now the main difference between the two way of filtering, and with this change we also preserve empty arrays.